pkg/descheduler/evictions/evictions.go

```diff
@@ -225,6 +239,14 @@ func (pe *PodEvictor) Evictable(opts ...func(opts *Options)) *evictable {
 			return nil
 		})
 	}
+	if options.nodeFit {
+		ev.constraints = append(ev.constraints, func(pod *v1.Pod) error {
+			if !nodeutil.PodFitsAnyOtherNode(pod, pe.nodes) {
```

Comment:
This will not fly for strategies like `PodLifeTime` for which it's valid to schedule the pod to the same node.

Reply:
I've now removed this switch from the `pod_lifetime` strategy.